health/health_log.c

```diff
@@ -67,6 +67,35 @@ inline void health_log_rotate(RRDHOST *host) {
     }
 }
 
+inline void health_label_log_save(RRDHOST *host) {
+    health_log_rotate(host);
+
+    if(likely(host->health_log_fp)) {
+        BUFFER *wb = buffer_create(1024);
+        netdata_rwlock_rdlock(&host->labels_rwlock);
+        struct label *l=localhost->labels;
+        while (l != NULL) {
+            buffer_sprintf(wb,"%s=%s\t ", l->key, l->value);
+            l = l->next;
+        }
+        netdata_rwlock_unlock(&host->labels_rwlock);
+
+        char *write = (char *) buffer_tostring(wb) ;
+
+        write[wb->len-2] = '\n';
+        write[wb->len-1] = '\0';
+
+        if (unlikely(fprintf(host->health_log_fp, "L\t%s"
+                , write
+        ) < 0))
+            error("HEALTH [%s]: failed to save alarm log entry to '%s'. Health data may be lost in case of abnormal restart.",
+                  host->hostname, host->health_log_filename);
+        else {
+            host->health_log_entries_written++;
+        }
```

Comment:
Please add `buffer_free(wb);`. The alternative is to make it static / single-initialization, but this is a very infrequent operation and we can probably save the memory instead.